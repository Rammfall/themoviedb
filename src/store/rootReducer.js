import { combineReducers } from 'redux'

import session from 'Store/concepts/session/reducer'
import account from 'Store/concepts/account/reducer'
import movies from 'Store/concepts/movies/reducer'
import data from 'Store/concepts/data/reducer'

import { RESET_STORE } from 'Store/types'

const reducers = combineReducers({
  session,
  account,
  movies,
  data
})

export default (state, action) => {
  switch (action.type) {
    case RESET_STORE:
      return reducers(undefined, action)
    default:
      return reducers(state, action)
  }
}
