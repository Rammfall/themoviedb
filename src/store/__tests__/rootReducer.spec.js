import rootReducer from 'Store/rootReducer'

describe('check root reducer', () => {
  it('matches snapshot', () => {
    expect(rootReducer).toMatchSnapshot()
  })
})
