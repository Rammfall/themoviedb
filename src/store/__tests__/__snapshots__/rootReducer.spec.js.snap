// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check root reducer matches snapshot 1`] = `[Function]`;
